src/test/java/com/stripe/model/PagingIteratorTest.java

```diff
@@ -168,4 +236,28 @@ void testAutoPaginationWithoutApiKey() throws StripeException {
           collection.autoPagingIterable();
         });
   }
+
+  @Test
+  void testAutoPaginationRequestOptionsPropagation() throws StripeException {
+    // set some arbitrary parameters so that we can verify that they're
+    // used for requests on ALL pages
+    final Map<String, Object> page0Params = new HashMap<>();
+    page0Params.put("foo", "bar");
+
+    Stripe.apiKey = null;
```

Comment:
nit: use the preserve/restore pattern for global settings https://github.com/stripe/stripe-java/blob/110ae567c68aa67cfe697a6309d35e3d13c5e419/src/test/java/com/stripe/net/StripeRequestTest.java#L125-L140

Reply:
This is already handled in the parent class:
https://github.com/stripe/stripe-java/blob/ba64a45bccbdafae779c841e64802941f6243e22/src/test/java/com/stripe/BaseStripeTest.java#L136